app.rb

```diff
@@ -0,0 +1,523 @@
+    current_authors = load_data('./store/authors.json') || []
+    authors_to_store = current_authors + @authors.map do |author|
+      {
+        'id' => author.id,
+        'first_name' => author.first_name,
+        'last_name' => author.last_name
+      }
+    end
+
+    store_data('./store/authors.json', authors_to_store)
+  end
+
+  def save_labels
+    current_labels = load_data('./store/labels.json') || []
+    labels_to_store = current_labels + @labels.map do |label|
+      {
+        'id' => label.id,
+        'title' => label.title,
+        'color' => label.color
+      }
+    end
+
+    store_data('./store/labels.json', labels_to_store)
+  end
+
+  def store_data(filename, data)
+    File.write(filename, JSON.generate(data))
+  rescue StandardError => e
+    puts "Error writing file: #{e}"
+  end
+
+  def save_genres
+    current_genres = load_data('./store/genres.json') || []
+    genres_to_store = current_genres + @genres.map do |genre|
+      {
+        'id' => genre.id,
+        'name' => genre.name
+      }
+    end
+
+    store_data('./store/genres.json', genres_to_store)
+  end
+end
+
+module LoadHandler
+  def load_all_items
+    @books = load_data('./store/books.json')
+    @music_albums = load_data('./store/music_albums.json')
+    @games = load_data('./store/games.json')
+    @genres = load_data('./store/genres.json')
+    @labels = load_data('./store/labels.json')
+    @authors = load_data('./store/authors')
+  end
+
+  def load_data(filename)
+    unless File.exist?(filename)
+      File.open(filename, 'a')
+      return []
+    end
+
+    data = File.read(filename)
+    JSON.parse(data) unless data.empty?
+  end
+end
+
+class App
+  include BookModule
+  include GameModule
+  include MusicAlbumModule
+  include ListHandler
+  include StoreHandler
+  include LoadHandler
+
+  def initialize
+    @genres = []
+    @books = []
+    @music_albums = []
+    @games = []
+    @authors = []
+    @labels = []
+    @items = []
+  end
+
+  def display_options
+    puts 'Please select an option:'
+    puts '  1. Create a new book'
+    puts '  2. Create a new music album'
+    puts '  3. Create a new game'
+    puts '  4. List all books'
+    puts '  5. List all music albums'
+    puts '  6. List all games'
+    puts '  7. List all genres'
+    puts '  8. List all labels'
+    puts '  9. List all authors'
+    puts ' 10. List all items'
+    puts ' 11. Load items from JSON file'
+    puts ' 12. Exit'
+    puts
+  end
+
+  def process_input(input)
+    case input
+    when 1 then create_book
+    when 2 then create_music_album
+    when 3 then create_game
+    when 4 then list_books
+    when 5 then list_music_albums
+    when 6 then list_games
+    when 7 then list_genres
+    when 8 then list_labels
+    when 9 then list_authors
+    when 10 then list_items
+    when 11 then load_all_items
+    when 12
+      save_genres
+      save_authors
+      save_labels
+      puts '  Thank you for using my app!.....Goodbye!'
+      exit
```

Comment:
- [ ] The app crashes when exiting the app, this could be the issue of saving and retrieving data from JSON files. Kindly fix this issue so that the app option 12 works as expected.



![exit](https://github.com/danielochuba/catalog-of-my-things/assets/53124548/ca8c7b66-182c-44f0-806b-144fac8d8dd1)

Reply:
https://github.com/danielochuba/catalog-of-my-things/assets/81446846/a494e4ab-05d2-4d82-850c-64545ba408d4

@elisha2kyakpo1 It works well on my machine